tempel.el

```diff
@@ -144,6 +144,21 @@ may be named with `tempel--name' or carry an evaluatable Lisp expression
     ((or 'n 'n> '> '& '% 'o) " ")
     (_ "_")))
 
+(defun tempel--expand-locations (locations)
+  "Return the list of files specified by LOCATIONS.
+LOCATIONS is a list of files and directories."
+  (setq locations (if (listp locations) locations (list locations)))
+  (let (files)
+    (dolist (file-or-dir locations)
+      (if (file-directory-p file-or-dir)
+	  (cl-callf append files (directory-files file-or-dir t))
```

Comment:
I am not fond of `cl-callf`, just write it out explicitly, it won't be much longer. It's only a minor stylistic thing.